### ISSUE TITLE
Add icon Magazine

### ISSUE THREAD

**Issue by KayleyHill** [lucide-icons/lucide#1317]
![magazine](https://github.com/lucide-icons/lucide/assets/135126101/0ddb47c8-bbe1-46e8-81f1-dc4de53fa0b5)

Magazine icon I have created from the newspaper icon (here again as I am not sure what fork branch to select for pull request)

**Comment by xnousnow** [comment]
Removed some useless points and optimized with svgomg

```svg
<svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" fill="none">
  <path stroke="#000" stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M7 3h11.2a2 2 0 0 1 2 2v11a2 2 0 0 1-2 2H7V4.1a2 2 0 0 0-2-2 2 2 0 0 0-2 2V18"/>
  <path stroke="#000" stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M7 18a2 2 0 0 0-2-2 2 2 0 0 0-2 2 4 4 0 0 0 4 4h9.5m-.2-8H11m0-7h5.3v3H11V7Z"/>
</svg>
```

**Comment by jguddas** [comment]
I optimized it and snapped a note ![icon (64)](https://github.com/lucide-icons/lucide/assets/25524993/671c1bf0-e95b-4027-bc37-4db3a845fa73)

https://lucide-site-git-feat-added-edit-page-jguddas.vercel.app/edit

```html
  <rect x="11" y="7" width="5.3" height="3" />
  <path d="M6.7 3h11.5a2 2 0 0 1 2 2v11a2 2 0 0 1-2 2H7V4.1a2 2 0 1 0-4 0V18" />
  <path d="M7 18a2 2 0 1 0-4 0 4 4 0 0 0 4 4h9.5" />
  <path d="M16.3 14H11" />
```

**Comment by xnousnow** [comment]
More pixel-perfecter I guess?
```svg
  <rect x="11" y="7" width="5.3" height="3" />
  <path d="M7 3h11a2 2 0 0 1 2 2v11a2 2 0 0 1-2 2H7V4.1a2 2 0 1 0-4 0V18" />
  <path d="M7 18a2 2 0 1 0-4 0 4 4 0 0 0 4 4h9.5" />
  <path d="M16.3 14H11" />
```

**Comment by jguddas** [comment]
The rect needs to at least be 4px in height due to the 2px gap rule, something like this would work.

```svg
  <rect x="11" y="6" width="6" height="4" />
  <path d="M5 2h14a2 2 0 0 1 2 2v12a2 2 0 0 1-2 2H7V4a2 2 0 1 0-4 0v14a4 4 0 0 0 4 4h10" />
  <path d="M3 18a2 2 0 1 1 4 0" />
  <path d="M17 14h-6" />
```

**Comment by jguddas** [comment]
There is also the suggestion to use round corners for rects, what do you think of something like this?
![icon (65)](https://github.com/lucide-icons/lucide/assets/25524993/99e03cd9-de3c-4611-9a27-8aa81ecec2fd)

**Comment by xnousnow** [comment]
Both are great! I'd prefer the first one. I'll wait for other members to review ✅

**Comment by KayleyHill** [comment]
I agree with @xnousnow, the rounded version is brilliant @jguddas, though depends if people mind it looking a bit different to the newspaper that it is based of. Thank you for improving and refining it everyone.

**Comment by KayleyHill** [comment]
@ericfennis  this was designed for my work which is magazine marketing. We use feather and now Lucide
 for our icons and my boss was not happy with the newspaper icon and wanted something more bespoke for magazines. So the use case is magazine/zine publishing.

**Comment by jguddas** [comment]
I think a newspaper and a magazine are distinct enough in practice and also when looking at this icon.

**Comment by ericfennis** [comment]
@KayleyHill check makes sense. This also nice alternative for a feed.

Feel free to create a PR.

**Comment by xnousnow** [comment]
@KayleyHill Note that you could import the icon as svg, meaning you don't really need to wait if you are ok about changing the code later when the icon is added into the package. Opening PR would be best option still!
